### ISSUE TITLE
`theme` function doesn't handle values without quotes well

### ISSUE THREAD

**Issue by RobinMalfait** [tailwindlabs/tailwindcss-intellisense#621]
**What version of Tailwind CSS IntelliSense are you using?**

Latest version on Tailwind Play

**What version of Tailwind CSS are you using?**

v3.1.8, insiders

**What package manager are you using?**

npm

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/NKLx1ffCmG?file=css

**Describe your issue**

While looking at this PR https://github.com/tailwindlabs/tailwindcss.com/pull/1370, I noticed that intellisense is causing an issue where if you omit the quotes, that it doesn't like that as much. It's implemented in Tailwind CSS itself and works there so therefore I think it is an intellisense bug.

Accepted:
```css
.foo {
  padding: theme('padding.4');
}
```

Not accepted:
```css
.foo {
  padding: theme(padding.4);
}
```

Might be interesting to look at this together with #614? 🤔

**Comment by davidjbng** [comment]
I did some digging how to solve this (not very promising unfortunately).
The behavior actually needs be changed in vscode-css-languageservice. It basically treats `theme` just like any other CSS function and does not accept `.`, `[` nor `]` characters as a valid argument (take a [look at this function for details](https://github.com/microsoft/vscode-css-languageservice/blob/main/src/parser/cssParser.ts#L1759-L1788)).

I'm not sure if this is a reasonable change to land there, since it also affects parsing of regular CSS functions.

**Comment by bradlc** [comment]
Fixed via 1b730cb6568d8e19b160fe25bdb0b6f8157d174c
